client/app/src/main/java/com/microsoft/research/karya/data/repo/WorkerRepository.kt

```diff
@@ -164,18 +167,40 @@ class WorkerRepository @Inject constructor(private val workerAPI: WorkerAPI, pri
 
   suspend fun getAllWorkers() =
     withContext(Dispatchers.IO) {
-      return@withContext workerDao.getAll()
+      updateWorkerCacheIfRequired()
+      return@withContext workerCache.values.toList()
     }
 
   suspend fun getWorkerById(id: String) =
     withContext(Dispatchers.IO) {
-      return@withContext workerDao.getById(id)
+      updateWorkerCacheIfRequired()
```

Comment:
Maybe  `updateWorkerCacheIfRequired`  can be shifted to a higher-order function to avoid duplication of its call?

Reply:
I don't see how a HOF wouuld help here, do you have any ideas?